### PR TITLE
docs: consolidate documentation

### LINKED ISSUE #91: Consolidate and audit documentation
Currently, our `README` is very light on details and acts as essentially a table of contents that links to our other documentation. This makes finding information more difficult, as you have to guess which file has the info you need, then scan it for info, rinse and repeat until you find it (or give up). While putting everything in the `README` would make it a very long read, it would be easier for developers to scan for information that they need. With the right heading structure and keeping the table of contents at the start, it should still be possible to skip to relevant sections.

While we're at it, we should audit the existing docs for accuracy and fill in any gaps that we find (or at least create issues to fill in those gaps).

### PR DESCRIPTION
**Pull request by dustin-jw**
## Description

This moves nearly all of the documentation into the `README` to make scanning for info easier for developers using the template. This also fixes some inaccuracies, adds some context about things that were under-documented, and adds info for people using the starter template for how to customize it for their projects.

While updating the docs about permalinks, I fixed an issue that broke pagination for custom URL structures, and I renamed the `sparkpress-standard.xml` rule definitions file to reduce the total number of things a developer would need to update when using the starter template. These changes are technically out of scope, but relatively small changes, so it seemed better to include them here rather than split them into different PRs where we'd have to update the docs multiple times.

Closes #91

## To Validate

1. Make sure all PR Checks have passed
2. Pull down this branch
3. Run `npm start`
4. Check that the updated pagination instructions work as expected, and you don't get a 404 response for page 2 of results
5. Change your `.vscode/settings.json` if you have one to point to the `phpcs-rules-standard.xml` file instead of `sparkpress-standard.xml` and confirm that you see linting warnings in VS Code if you intentionally introduce an issue
6. Check the updated `README`, `CONTRIBUTING.md`, and `CODE_OF_CONDUCT.md` files for accuracy and thoroughness

### CODE REVIEW

**Review by rise-erpelding — CHANGES_REQUESTED** [review]
This is great! There were a few minor changes in the `README` that I pointed out that would need to be fixed, and I'll probably take another look through just to make sure there wasn't anything else that stands out.

Linting seems to work ok after changing the name of the php standards file, but running `npm run lint` keeps stopping my Docker container whereas it doesn't for me on `main` (with the previous `settings.json` settings). Does this happen for you too? I am a little puzzled as to why this might be happening.

Otherwise, all looks good!

File: README.md [hunk 1]
```diff
@@ -1,23 +1,520 @@
...
+This project requires [Docker][docker] and [Node.js][node] for local development. You may also find it useful to install [Composer][composer] for linting in your editor, but it isn't strictly necessary. To run the project for the first time, do the following:
+
+1. Duplicate `.env.example` and rename it `.env`, changing variables [as needed](#setting-local-environment-variables)
+1. Run `npm install`
+1. Run `npm run plugins:install` (this is necessary for the plugin for custom blocks)
+1. Either run `npm start` or run `npm run build:dev` and `npm run serve:dev` in separate terminals
+1. Based on whether you have a database to import or not, do one of the following:
+   - Visit `https://localhost:8000/wp-admin` and run through the WordPress setup
+   - Follow the instructions for [importing a DB](#importing-databases)
+1. Go to `http://locahost:8000` to ensure the home page loads
```
Comment (rise-erpelding) [inline README.md:49]:
```suggestion
1. Go to `http://localhost:8000` to ensure the home page loads
```

File: README.md [hunk 2]
```diff
@@ -1,23 +1,520 @@
...
+# run JS tests in watch mode
+npm run test:watch
+
+# run JS tests and get a coverage report
+npm run test:ci
+```
+
+### Sass (`scss/`)
+
+The `scss/` directory holds our styles for the site which are written using [Sass][sass] and follows a modified [BEMIT][bemit] approach to naming and organizing. `stylelint` is used to check that SCSS files conform to our standards. Run `npm run lint:scss` to check SCSS files for issues.
```
[the remark this answers is not in the document]
Comment (dustin-jw) [reply README.md:223]: I think it was missing from the original docs too, now that I look at it, but added one!

File: README.md [hunk 3]
```diff
@@ -1,23 +1,520 @@
...
+- `PRODUCTION_REPO` - The address of the Pantheon remote repo. See instructions below for [obtaining the value](#obtaining-production_repo-value).
+- `PRODUCTION_USER_EMAIL` - The email address that will be used for deployment commit messages.
+- `PRODUCTION_USER_NAME` - The username that will be used for deployment commit messages.
+- `KNOWN_HOSTS` - Required in order to authenticate with Pantheon. See instructions below for [obtaining the value](#obtaining-known_hosts-value).
+
+##### Obtaining `PRODUCTION_REPO` value
+
+1. Visit your Pantheon [site dashboard](https://docs.pantheon.io/guides/account-mgmt/workspace-sites-teams/sites#site-dashboard).
+1. In the "Development Mode" section, click "Clone with git" and copy the value.
+   ![Pantheon Development Mode](_assets/pantheon-development-mode.png)
```
[the remark this answers is not in the document]
Comment (dustin-jw) [reply README.md]: It was, didn't realize it was included twice, good catch!

**Comment by dustin-jw** [comment]
> Linting seems to work ok after changing the name of the php standards file, but running `npm run lint` keeps stopping my Docker container whereas it doesn't for me on `main` (with the previous `settings.json` settings). Does this happen for you too? I am a little puzzled as to why this might be happening.

Turns out the `--abort-on-container-exit` flag was the issue, but we don't actually need it with the `|| docker compose down` that got added after the flag.

File: src/php/functions.php [hunk 4]
```diff
@@ -123,3 +123,18 @@ function filter_password_protected_posts( $query ) {
 	return $query;
 }
 add_filter( 'pre_get_posts', 'filter_password_protected_posts' );
+
+/**
+ * Normalize queries for paginated archive pages to support custom permalink structures.
+ *
+ * @param array $query_string - The incoming query string to be normalized.
+ */
+function normalize_pagination_query_stings( $query_string ) {
```
Comment (jonoliver) [inline src/php/functions.php:132]:
```suggestion
function normalize_pagination_query_strings( $query_string ) {
```

File: src/php/functions.php [hunk 5]
```diff
@@ -123,3 +123,18 @@ function filter_password_protected_posts( $query ) {
...
+	if ( isset( $query_string['name'] ) && 'page' === $query_string['name'] && isset( $query_string['page'] ) ) {
+		unset( $query_string['name'] );
+		$query_string['paged'] = $query_string['page'];
+	}
+
+	return $query_string;
+}
+add_filter( 'request', 'normalize_pagination_query_stings' );
```
Comment (jonoliver) [inline src/php/functions.php:140]:
```suggestion
add_filter( 'request', 'normalize_pagination_query_strings' );
```
This typo stings! 🐝

File: README.md [hunk 6]
```diff
@@ -1,23 +1,520 @@
...
+
+If you use Atom, go to Preferences > Packages. Open the `language-php` Core Package settings. Go to the Tab Type setting and set the drop down option to `hard`.
+
+#### VS Code
+
+If you use Microsoft VS Code, create a `settings.json` inside a `.vscode` directory at the root of the project. Include this in your setting (it will help make developing with PHP_Codesniffer much easier):
+
+```json
+{
+	"phpcs.standard": "wp-configs/sparkpress-standard.xml",
```
[the remark this answers is not in the document]
Comment (dustin-jw) [reply README.md]: Updated in https://github.com/sparkbox/sparkpress-wordpress-starter/pull/104/commits/ce088f9af0eb44bf6b66f313b2899371275da644

**Review by rise-erpelding — APPROVED** [review]

**Pull request merged**